apps/researcher/src/app/[locale]/page.tsx

```diff
@@ -32,17 +33,39 @@ import {ElementType} from 'react';
 // Revalidate the page every n seconds
 export const revalidate = 60;
 
-interface FacetProps {
+interface FilterSetting {
   name: keyof SearchResult['filters'];
   searchParamType: SearchParamType;
+}
+
+interface FacetComponent {
+  name: string;
   Component: ElementType;
+  customProps?: Record<string, unknown>;
 }
 
-const facets: ReadonlyArray<FacetProps> = [
-  {name: 'owners', searchParamType: 'array', Component: FilterSet},
-  {name: 'types', searchParamType: 'array', Component: FilterSet},
-  {name: 'subjects', searchParamType: 'array', Component: FilterSet},
-  {name: 'publishers', searchParamType: 'array', Component: FilterSet},
+const filterSettings: ReadonlyArray<FilterSetting> = [
+  {name: 'owners', searchParamType: 'array'},
+  {name: 'types', searchParamType: 'array'},
+  {name: 'subjects', searchParamType: 'array'},
+  {name: 'publishers', searchParamType: 'array'},
+  {name: 'dateCreatedStart', searchParamType: 'number'},
+  {name: 'dateCreatedEnd', searchParamType: 'number'},
+];
+
+const facetComponents: ReadonlyArray<FacetComponent> = [
+  {name: 'owners', Component: FilterSet},
+  {
+    name: 'dateCreated',
+    Component: DateRangeFilter,
```

Comment:
Would it be possible to align the names of the components? The existing component for facets is called `FilterSet`, the new component for date ranges is called `DateRangeFilter`. Are these both 'filter components'? Would names such as `SelectFilter` (or `MultiSelectFilter`) and `DateRangeFilter` make sense?

Reply:
Perfect!